corehq/apps/hqmedia/tasks.py

```diff
@@ -213,13 +221,18 @@ def create_files_for_ccz(build, build_profile_id, include_multimedia_files=True,
 
     # Don't rebuild the file if it is already there
     if not (os.path.isfile(fpath) and settings.SHARED_DRIVE_CONF.transfer_enabled):
-        files, errors, file_count = _build_ccz_files(build, build_profile_id, include_multimedia_files,
-                                                     include_index_files, download_id, compress_zip,
-                                                     filename, download_targeted_version)
-        file_cache = _zip_files_for_ccz(fpath, files, current_progress, file_progress, file_count, compression,
-                                        task)
+        with build.timing_context("_build_ccz_files"):
```

Comment:
I don't love this, for two reasons - It adds clutter, and unlike `@time_method`, this won't noop if the parent timer is inactive.  I assume the overhead of these is small enough that that's not worth worrying about, but I'd still be interested in alternative approaches.  Here's what came to mind:

1. Restructure `create_files_for_ccz` and the associated helper functions as a class, and use `time_method`.  That'd solve both issues, but it's a big refactor that I'm not sure is justified. 
2. Make a `@time_function` decorator that can fetch a timing context from one of the function args.  I actually did this, but then each function needs access to the timing context, and most of these don't currently have it (when only their outer scope needs it at the moment)
3. Modify `TimingContext` in some way so as not to start if the root or parent timer wasn't started.  Again, not sure this saves any appreciable computation time.

Reply:
> this won't noop if the parent timer is inactive

What does happen if the parent timer is inactive?

I agree this adds some clutter, but I can live with it. Definitely don't think restructuring `create_files_for_ccz` is necessary.

Reply:
It doesn't look at the parent timer at all, so this calls `time.time()` on each `__enter__` and `__exit__` and saves it to the timer object.